src/main/java/de/rytrox/varo/teams/spawnpoint/SpawnPointService.java

```diff
@@ -0,0 +1,92 @@
+
+    private final Varo main;
+
+    private final TeamMemberRepository teamMemberRepository;
+
+    public SpawnPointService(@NotNull Varo main, @NotNull TeamMemberRepository teamMemberRepository) {
+        this.main = main;
+        this.teamMemberRepository = teamMemberRepository;
+    }
+
+    public void setSpawnPoint(Player executor, String targetName) {
+        Bukkit.getScheduler().runTaskAsynchronously(main, () -> {
+            // get Target
+            TeamMember member = teamMemberRepository.getPlayer(Bukkit.getOfflinePlayer(targetName));
+
+            if(member != null) {
+                SpawnPoint point = Optional.ofNullable(member.getSpawnPoint())
+                        .orElse(new SpawnPoint());
+
+                point.setLocation(executor.getLocation());
+                member.setSpawnPoint(point);
+
+                // If a player has a spawnpoint, it is registered
+                if(member.getStatus() == PlayerStatus.NOT_REGISTERED)
+                    member.setStatus(PlayerStatus.ALIVE);
+
+                main.getDB().update(member);
+                executor.sendMessage(ChatColor.translateAlternateColorCodes('&',
+                        String.format("&8[&6Varo&8] &7Du hast den Spawnpoint von %s erfolgreich gesetzt!", member.getOfflinePlayer().getName())));
+                main.getLogger().log(Level.INFO, String.format("Player %s sets Spawnpoint of %s to %s", executor.getName(), member.getOfflinePlayer().getName(), point));
+            } else executor.sendMessage(ChatColor.RED + "Dieser Spieler ist nicht bekannt");
+        });
+    }
+
+    /**
+     * Teleport the player when the player joins and the game is running
+     *
+     * @param event the TeamMemberSpawnEvent
+     */
+    @EventHandler
+    public void onSpawn(TeamMemberSpawnEvent event) {
+        Player player = event.getPlayer();
+
+        // Only teleport if the game is running and the player is alive
+        if(event.getMember().getStatus() == PlayerStatus.ALIVE &&
+                (gameStateHandler.getCurrentGameState() == GameStateHandler.GameState.MAIN ||
+                 gameStateHandler.getCurrentGameState() == GameStateHandler.GameState.FINAL)) {
+            Bukkit.getScheduler().runTask(main, () -> {
+                SpawnPoint spawnPoint = event.getMember().getSpawnPoint();
+
+                if(spawnPoint != null) {
+                    player.teleport(spawnPoint.getLocation());
+                }
+            });
+        }
+    }
+
+    @EventHandler
+    public void onUpdateSpawnPoint(TeamMemberDisconnectEvent event) {
+        if(gameStateHandler.getCurrentGameState() == GameStateHandler.GameState.MAIN &&
+                event.getMember().getStatus() == PlayerStatus.ALIVE &&
+                event.getMember().getSpawnPoint() != null) {
+            event.getMember().getSpawnPoint().setLocation(event.getPlayer().getLocation());
+
+            main.getDB().update(event.getMember().getSpawnPoint());
```

Comment:
Hier müsste es nicht nur MAIN sondern auch START und FINAL sein. Wobei ich da den Sinn nicht verstehe, warum du den Spawnpoint setzt, wenn das doch vom Server sowieso gemacht wird... 
Die Spawnpoints in der DB sind doch nur für die initialen Spawns vor Beginn gedacht

Reply:
Siehe Kommentar dadrüber. Ich dachte mir es wäre sinnvoller, immer den aktuellen SpawnPoint während des Spiels in der Datenbank zu speichern, um bei BorderEvents zu detecten, ob OfflinePlayers out of Range sind